### PR TITLE
Feature/apartment carosel

### PR DESCRIPTION
**Pull request by KShih**
This PR includes:
1. Introduce multiple image field to location model with foreign key
2. modify apartment upload
3. modify form processing in location/views.py
4. modify refresh_apartment view.py to fetch multiple image
5. Introduce carousel to apartment.html
6. Introduce carousel preview thumbnail

image

Updated 11:39am, 12/10
image
image

### CODE REVIEW

File: location/models.py [hunk 1]
```diff
@@ -142,6 +142,20 @@ def picture_url(self):
...
+    @property
+    def picture_url(self):
+        if not self.image:
+            return None
+        elif "http" not in self.image.url:
+            return self.image.url
+        else:
+            return self.image
+
+
```
Comment (ab7289) [inline location/models.py]: would it be possible to get some test cases to cover this?

Screen Shot 2019-12-09 at 9 29 49 PM

**Comment by ab7289** [comment]
Overall I think this looks really awesome!!

Couple small things.

Would it be possible to move the thumbnail just below the main image? I think that would make it a bit more obvious that there are other images.

Screen Shot 2019-12-09 at 9 36 43 PM

Similarly, would it be possible to include the thumbnails just below the image on the apartment view page? I think this would make it more obvious to users that there are/aren't more images for them to check out.

Screen Shot 2019-12-09 at 9 37 27 PM

This may not be feasible in the time we have, but would it be possible instead of having a set number of fields on the apartment upload form to have a little + button that would add another image upload field to the form? or even just a single multi-upload field (if that exists)?

Screen Shot 2019-12-09 at 9 39 30 PM

Lastly, if that is possible, could we get that on the apartment update form as well?

These changes look great!

File: location/models.py (hunk 1 above)
Comment (KShih) [reply location/models.py]: @ab7289 Updated!

**Comment by ab7289** [comment]
@KShih , would it be possible to give the small thumbnails a fixed size so they don't overlap the bottom of the container?

Screen Shot 2019-12-10 at 12 34 38 PM

**Pull request merged**